src/main/java/io/cdap/plugin/gcp/spanner/source/SpannerSource.java

```diff
@@ -192,7 +195,8 @@ private void initializeConfig(Configuration configuration, String projectId) {
     setIfValueNotNull(configuration, SpannerConstants.SERVICE_ACCOUNT_FILE_PATH, config.getServiceAccountFilePath());
     setIfValueNotNull(configuration, SpannerConstants.INSTANCE_ID, config.instance);
     setIfValueNotNull(configuration, SpannerConstants.DATABASE, config.database);
-    setIfValueNotNull(configuration, SpannerConstants.QUERY, String.format("Select * from %s;", config.table));
+    setIfValueNotNull(configuration, SpannerConstants.QUERY, Strings.isNullOrEmpty(config.importQuery) ?
+      String.format("Select * from %s;", config.table) : config.importQuery);
```

Comment:
1.) Is `;` needed at the end?
2.) `Select` should have `S` capitalized? Is the query case sensitive?
3.) What happens when table name is case sensitive? Is it supported?

Reply:
@CuriousVini this was how it was originally. As for the questions: